Define terms 'Out-of-band' and 'In-band' (#127)
tc39/how-we-work: terminology.md

Additionally, in a shorter-term sense, it could overwhelm developers to have too many syntax-heavy features ship in too narrow of a timeframe.

## In-Band

### Definition

Used mainly in opposition to [Out of band](#out-of-band), denoting that all configuration of a feature or a behavior is done within the context of ECMASCript source.

### Example

> - The stage 3 Import assertions proposal assert pragma

Reviewer: Another possible nice example:
- the `"use strict"` directive is an in-band way to define the "language mode"
- script vs module is defined by out-of-band configuration (`type="module"` in HTML, or `.mjs`/`"type":"module"` in Node.js)

Author: Thank you for additional examples!  Will add that in

Reviewer 2: nit: `.mjs` / `.js` + `"type":"module"` in node.js

Additionally, in a shorter-term sense, it could overwhelm developers to have too many syntax-heavy features ship in too narrow of a timeframe.

## In-Band

### Definition

Used mainly in opposition to [Out of band](#out-of-band), denoting that all configuration of a feature or a behavior is done within the context of ECMASCript source.

### Example

- The stage 3 Import assertions proposal assert pragma

> - The "use strict" directive, an in-band way to define the semantics of the script contained in the scope where it is used.

Reviewer 3: ```suggestion
- The "use strict" directive, an in-band way to enable strict mode for some portion of code.
```

Also I would probably put this first since readers will be much more familiar with it.

### References

[Import assertions proposal FAQ](https://github.com/tc39/proposal-import-assertions/blob/ae28137f45f6acd7fc61be4f7193759570a776ff/README.md#why-not-out-of-band) in opposition to out-of-band

## Out-of-band

### Definition

Used to denote a behavior that applies to the execution of a ECMAScript module or script that is configured outside of ECMAScript source text.

### Example

An example of out-of-band configuration is Import Maps.  With this feature, the module specifier used in an import statement or the dynamic import function is interpreted by the host differently based on the import map configuration that is defined within a non-ECMAScript source.  This may be a HTML `<script>` tag with the `importmap` type in the browser, or in a configuration file like `package.json` in a Node.js project.

> Another example is whether an ECMAScript source file will be executed as a script or a module based on the `type` attribute of the `<script>` tag that caused the User Agent to fetch and load it within a browser context.  The same is done in Node.js throug the use of the .mjs extension or by specifying that the package is of type `"module"` in the `pacakge.json` file.

Reviewer 3: ```suggestion
Another example is whether an ECMAScript source file will be executed as a script or a module based on the `type` attribute of the `<script>` tag that caused the User Agent to fetch and load it within a browser context.  The same is done in Node.js through the use of the .mjs extension or by specifying that the package is of type `"module"` in the `pacakge.json` file.
```
